### ISSUE TITLE
Un/installer: "update to" version not respected

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/wordpoints#681]
The `update()` method allows both the version being updated from and the version being updated to to be specified. However, only the version being updated from is taken into account when assembling the list of updates to run. All updates greater than the from version will be performed, not just those up to the to version. The reason that we've never discovered this is because we are always specifying the latest version as the to version during normal operation, so it is expected for all of the updates greater than the from version to be run.

**Comment by JDGrimes** [comment]
In addition, it looks like the from version is optional, but that we assume that it is set when we run the version comparison. We check if it is empty, and if so assign the value from the database to `$this->update_from`, but then we don't use that when comparing the versions.

**Comment by JDGrimes** [comment]
This reveals an issue that we didn't previously have to deal with. When we check if the to version is >= to the update version, we are fine as far as releases go. The current version will the the to version (2.4.0, for example), and the update version would be something like 2.4.0 as well. However, during alpha/beta, the current version is only something like 2.4.0-alpha-1, which is not >= 2.4.0. The result is that the update wouldn't run, not until the version was upped to 2.4.0 at release time.

The solution would be to mark the update as being for 2.4.0-alpha-2, instead of just 2.4.0.

This reminded me of #420, where we originally made it possible to specify alpha/beta versions for updates. The concern there was more about supplying updates in a way that would not break sites that had been running the beta versions of the plugin. Either there has been a regression of sorts since then, or at the time we did not have to supply beta-style versions for updates for the sake of our tests, but only for end users. Now we do have to supply these versions for the updates both for the updates and for end users running beta versions.

In short, using beta-style versions for updates has already been a good idea for the sake of users running nightlies, and now it will be necessary across the board.

The only potential issue is that this change will affect more than just core, although some extensions might not otherwise find it necessary to use beta-style update versions otherwise, if they don't release beta versions. It shouldn't cause any issues for end-users though, only possibly a change in future versions that are being developed, for the sake of testing (if they need to test updates before upping the version number to the next stable release).
